RegenerateServiceKey/handler.ts

```diff
@@ -0,0 +1,146 @@
+
+import { Context } from "@azure/functions";
+import { identity } from "fp-ts/lib/function";
+import {
+  fromLeft,
+  taskEither,
+  TaskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import { ServiceModel } from "io-functions-commons/dist/src/models/service";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredBodyPayloadMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_body_payload";
+import { SubscriptionKeys } from "../generated/definitions/SubscriptionKeys";
+import { SubscriptionKeyTypePayload } from "../generated/definitions/SubscriptionKeyTypePayload";
+import { APIClient } from "../utils/clients/admin";
+import {
+  ErrorResponses,
+  IResponseErrorUnauthorized,
+  toErrorServerResponse
+} from "../utils/responses";
+
+type ResponseTypes =
+  | IResponseSuccessJson<SubscriptionKeys>
+  | IResponseErrorUnauthorized
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorNotFound
+  | IResponseErrorTooManyRequests
+  | IResponseErrorInternal;
+
+/**
+ * Type of a GetRegenerateServiceKeyHandler handler.
+ *
+ * GetRegenerateServiceKey expects a service ID and a subscriptionKeyType as input
+ * and returns regenerated subscriptionkeys as outcome
+ */
+type IGetRegenerateServiceKeyHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  attrs: IAzureUserAttributes,
+  serviceId: string,
+  subscriptionKeyTypePayload: SubscriptionKeyTypePayload
+) => Promise<ResponseTypes>;
+
+const regenerateServiceKeyTask = (
+  apiClient: ReturnType<APIClient>,
+  serviceId: string,
+  subscriptionKeyTypePayload: SubscriptionKeyTypePayload
+): TaskEither<ErrorResponses, IResponseSuccessJson<SubscriptionKeys>> =>
+  tryCatch(
+    () =>
+      apiClient.RegenerateSubscriptionKeys({
+        service_id: serviceId,
+        subscriptionKeyTypePayload
+      }),
+    errs => ResponseErrorInternal(JSON.stringify(errs))
+  ).foldTaskEither(
+    err => fromLeft(err),
+    maybeResponse =>
+      maybeResponse.fold(
+        errs => fromLeft(ResponseErrorInternal(JSON.stringify(errs))),
+        responseType =>
+          responseType.status !== 200
+            ? fromLeft(toErrorServerResponse(responseType))
+            : taskEither.of(ResponseSuccessJson(responseType.value))
+      )
+  );
+
+/**
+ * Handles requests for upload a service logo by a service ID and a base64 logo' s string.
+ */
+export function GetRegenerateServiceKeyHandler(
+  apiClient: ReturnType<APIClient>
+): IGetRegenerateServiceKeyHandler {
+  return (_, __, ___, ____, serviceId, subscriptionKeyTypePayload) => {
+    return regenerateServiceKeyTask(
+      apiClient,
+      serviceId,
+      subscriptionKeyTypePayload
+    )
+      .fold<ResponseTypes>(identity, identity)
+      .run();
+  };
+}
+
+/**
+ * Wraps a GetRegenerateServiceKey handler inside an Express request handler.
+ */
+export function GetRegenerateServiceKey(
```

Comment:
```suggestion
export function RegenerateServiceKey(
```

Reply:
Fixed in [98db5c7](https://github.com/pagopa/io-functions-services/pull/68/commits/98db5c77eefe2f2bbce99559596e00feed83aa0c)